src/useradd.c

```diff
@@ -2073,6 +2083,37 @@ static void create_home (void)
 			strcat (path, "/");
 			strcat (path, cp);
 			if (access (path, F_OK) != 0) {
+				/* Check if parent directory is BTRFS, fail if requesting
+				   subvolume but no BTRFS. The paths cound be different by the
+				   trailing slash
+				 */
+#if WITH_BTRFS
+				if (subvolflg && (strlen(prefix_user_home) - (int)strlen(path)) <= 1) {
```

Comment:
Why do you cast to `int` here?

Reply:
This is just an attempt at avoiding unsigned undeflow. I didn't check if this is impossible, but just in case. :crossed_fingers: